library/core/src/slice/mod.rs

```diff
@@ -865,8 +865,10 @@ impl<T> [T] {
     pub fn chunks_exact(&self, chunk_size: usize) -> ChunksExact<'_, T> {
         assert_ne!(chunk_size, 0);
         let rem = self.len() % chunk_size;
-        let len = self.len() - rem;
-        let (fst, snd) = self.split_at(len);
+        let fst_len = self.len() - rem;
+        // The following is the same as self.split_at(fst_len) but avoids a bounds check
+        // SAFETY: 0 <= fst_len < self.len() by construction above
+        let (fst, snd) = unsafe { (self.get_unchecked(0..fst_len), self.get_unchecked(fst_len..)) };
         ChunksExact { v: fst, rem: snd, chunk_size }
```

Comment:
Is `snd` second? If it is why isn't it `scd`?

Reply:
It is, and I don't know :) I'm fine with renaming these but that's how they were named before and also elsewhere in the same file, so if someone wants to have them renamed let's do it consistently at least everywhere in the same file.

Reply:
`fst` and `snd` are well-established names for these in functional programming langauges

* https://en.wikibooks.org/wiki/Haskell/Lists_and_tuples#Example:_fst_and_snd
* https://docs.racket-lang.org/plait/Predefined_Functions_and_Constants.html#%28def._%28%28lib._plait%2Fmain..rkt%29._fst%29%29
* https://www.cs.cornell.edu/courses/cs3110/2009sp/recitations/rec02.html

Amusingly, they are a more obvious alternative to [`car` and `cdr`](https://en.wikipedia.org/wiki/CAR_and_CDR)

Reply:
Ah, I am not coming from haskell so I it took me a while to read the context and guess that it is `second`. I would rather do `a`, `b` and `c` to show `fst`, `snd` and `trd` (hopefully I am correct on this). Heh, thanks for explaining I think I got the pattern now. `fth`, `fth`, `sth`, `sth`, `sth`, `eth`, `nth`, `tth`...